### PR TITLE
Maintain the high end of the 'roundoff domain' in both float and double precision

### PR DESCRIPTION
**Pull request by atmyers**
This way we set the proper value for the high end when mixed precision is used.

The proposed changes:
- [x] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Src/Base/AMReX_Geometry.H [hunk 1]
```diff
@@ -67,6 +67,32 @@ public:
     int coord;
 };
 
+    namespace detail {
+        template <typename T>
+        T bisect_prob_hi (amrex::Real plo, amrex::Real phi, amrex::Real idx, int ilo, int ihi, amrex::Real tol) {
+            T hi = static_cast<T>(phi - tol);
+            bool safe;
+            {
+                int i = int(Math::floor(hi - plo)*idx) + ilo;
```
Comment (WeiqunZhang) [inline Src/Base/AMReX_Geometry.H]: Should it be `floor((hi-plo)*idx)` instead of `floor(hi-pol)*idx`?

Comment (atmyers) [inline Src/Base/AMReX_Geometry.H]:
```suggestion
                int i = int(Math::floor((x - plo)*idx)) + ilo;
```

Comment (atmyers) [reply Src/Base/AMReX_Geometry.H]: Yes, good catch.

File: Src/Base/AMReX_Geometry.H [hunk 2]
```diff
@@ -67,6 +67,32 @@ public:
     int coord;
 };
 
+    namespace detail {
+        template <typename T>
+        T bisect_prob_hi (amrex::Real plo, amrex::Real phi, amrex::Real idx, int ilo, int ihi, amrex::Real tol) {
+            T hi = static_cast<T>(phi - tol);
+            bool safe;
+            {
+                int i = int(Math::floor((x - plo)*idx)) + ilo;
```
Comment (atmyers) [inline Src/Base/AMReX_Geometry.H]:
```suggestion
                int i = int(Math::floor((hi - plo)*idx)) + ilo;
```

**Review by WeiqunZhang — APPROVED** [review]

**Comment by ax3l** [comment]
I'll merge this since it's approved & tests pass :)

**Comment by ax3l** [comment]
X-ref:
- https://github.com/ECP-WarpX/WarpX/issues/3155
- https://github.com/AMReX-Codes/amrex/pull/2679#issuecomment-1079283896

**Comment by ax3l** [comment]
cc @PhilMiller re #2814

**Comment by ax3l** [comment]
@KZhu-ME on Slack:
> It fixes issue [3155](https://github.com/ECP-WarpX/WarpX/issues/3155) for us! Thanks!

**Pull request merged**